paderbox/utils/misc.py

```diff
@@ -74,3 +53,67 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         sys.stdout.close()
         sys.stdout = self._original_stdout
+
+
+def all_equal(x: Iterable[Hashable]) -> bool:
+    """
+    Checks if all elements in `x` are equal. Returns `False` if `x` is empty.
+
+    Defined to improve readability.
+
+    Examples:
+        >>> all_equal((1, 1, 1, 1))
+        True
+        >>> all_equal((1, 1, 1, 2))
+        False
+    """
+    return len(set(x)) == 1
+
+
+def all_unique(x: Iterable[Hashable]) -> bool:
+    """
+    Checks if all elements in `x` are unique. Returns `True` if `x` is empty.
+
+    Defined to improve readability.
+
+    Examples:
+        >>> all_unique((1, 2, 3, 4))
+        True
+        >>> all_unique((1, 2, 3, 1))
+        False
+    """
+    return len(set(x)) == len(list(x))
+
+
+def all_in(x: Iterable[Hashable], y: Iterable[Hashable]) -> bool:
+    """
+    Check if all elements in `x` are in `y`. Returns `True` if `x` is empty.
+
+    Equivalent to `set(x).issubset(y)`, but faster.
+
+    Defined to improve readability.
+
+    Examples:
+        >>> all_in([1, 2, 2, 1, 2], [1, 2, 3])
+        True
+        >>> all_in([1, 2, 2, 4, 2], [1, 2, 3])
+        False
+    """
+    return all(x_ in y for x_ in x)
+
+
+def any_in(x: Iterable[Hashable], y: Iterable[Hashable]) -> bool:
```

Comment:
Is this not an intersection check?

Reply:
Yes, it is, do you think `intersects_with` is a better name? This could be very confusing

Reply:
I like the idea of keeping the names of these functions similar. I would keep the any/all prefix

Reply:
My idea was to rename both function :).
I am not sure, that `intersects_with` is the best naming. It sounds like a non-symmetric operation.
Maybe something like `has_intersection`?

Reply:
I chose the names `all_in` and `any_in` as extensions of Python's `in` operator, so that, for example,
```python
assert [o in available_options for o in options]
```
becomes more obvious
```python
assert all_in(options, available_options)
```
I don't think that `has_intersection` is more obvious in this case. But there are other cases where it is.

Reply:
I ask just because, I prefer to use existing python names, when they exist.
Since `all_in` is `set.issubset`, where we allow more types, I thought it would be better to align the names.

For `any_in` it is a little different, because it needs two functions (`set.intersection` and `bool`).

I think in your example, both are easy to interpret:

`assert all_in(options, available_options)`

vs

`assert issubset(options, available_options)`